src/Pact/Gas.hs

```diff
@@ -15,85 +15,92 @@ module Pact.Gas
  , putGas)
  where
 
+import Control.Monad(when)
 import Data.Text
 import Data.IORef
+import Pact.Types.Gas
 import Pact.Types.Runtime
 import Control.Lens
 import Control.Arrow
 import Data.Word
 import Pact.Types.Pretty
 
 -- | Compute gas for some application or evaluation.
-computeGas :: Either (Info,Text) FunApp -> GasArgs -> Eval e Gas
+computeGas :: Either (Info,Text) FunApp -> GasArgs -> Eval e ()
```

Comment:
Regarding this change, and every signature below this that no longer threads gas: we weren't using the gas output from this signature at all.

We can use this `MicroGas` PR to remove the needless gas threading. See the change to `call` in `Eval`.

Reply:
I'd vote for renaming `computeGas` → `updateGas` or `checkGas` or something like that (and similarly for other `compute*` functions) — looks like it's closer to what the function is doing.

Reply:
Good suggestion @0xd34df00d

Reply:
I'm actually against this, mostly because it changes a bunch of callsites (making the diff bigger for no reason) and because the difference between "it computes gas" and "it computes and tallies gas" is minor.